### PR TITLE
#1120 Update to add value_var parameter to derive_vars_atc()

### PR DESCRIPTION
**Pull request by tamarasenior**
Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiral/articles/development_process.html) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `devel` branch until you have checked off each task.

- [x] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/) 
- [x] Updated relevant unit tests or have written new unit tests - See [Unit Test Guide](https://pharmaverse.github.io/admiral/articles/unit_test_guidance.html#writing-unit-tests-in-admiral-)
- [x] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiral/articles/programming_strategy.html#deprecation-1)?
- [x] Update to all relevant roxygen headers and examples 
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [x] Address any updates needed for vignettes and/or templates
- [x] Update `NEWS.md` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers)
- [x] Build admiral site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the "[Reference](https://pharmaverse.github.io/admiral/reference/index.html)" page. 
- [x] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [x] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [x] Link the issue in the Development Section on the right hand side.
- [x] Address all merge conflicts and resolve appropriately 
- [x] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

File: R/derive_vars_transposed.R [hunk 1]
```diff
@@ -154,17 +157,19 @@ derive_vars_transposed <- function(dataset,
...
+                            value_var = FASTRESC) {
+  value_var <- assert_symbol(enquo(value_var))
   assert_vars(by_vars)
   assert_data_frame(dataset, required_vars = replace_values_by_names(by_vars))
   assert_data_frame(dataset_facm, required_vars = vars(!!!by_vars, FAGRPID, FATESTCD, FASTRESC))
 
   dataset %>%
     derive_vars_transposed(
-      select(dataset_facm, !!!unname(by_vars), FAGRPID, FATESTCD, FASTRESC),
+      select(dataset_facm, !!!unname(by_vars), !!unname(value_var), FAGRPID, FATESTCD, FASTRESC),
```
Comment (bundfussr) [inline R/derive_vars_transposed.R:169]: `FASTRESC` should be removed.

File: R/derive_vars_transposed.R [hunk 2]
```diff
@@ -105,13 +105,16 @@ derive_vars_transposed <- function(dataset,
 #'
 #' @param dataset_facm FACM dataset
 #'
-#'   The variables specified by the `by_vars` parameter, `FAGRPID`, `FATESTCD` and
+#'   The variables specified by the `by_vars` and `value_var` parameters, `FAGRPID`, `FATESTCD` and
 #'   `FASTRESC` are required
```
Comment (bundfussr) [inline R/derive_vars_transposed.R:109]: `FASTRESC` should be removed.

**Review by bundfussr — CHANGES_REQUESTED** [review]

**Comment by rossfarrugia** [comment]
@tamarasenior please can you make the requested updates to this PR. we need it merged for Roche ADCM to run

**Comment by tamarasenior** [comment]
@rossfarrugia @bundfussr I've implemented the requested changes, but there is a broken link in devel that is causing one of the checks to fail

**Comment by rossfarrugia** [comment]
@tamarasenior looks like https://r-pkgs.org/git.html is now a broken link in vignettes/git_usage.Rmd and i cant find an alternative left on that page. Can you delete the row as part of this PR please and that should solve it?

@thomas-neitmann @bms63 tagging as FYI in case you want the link replaced.

File: R/derive_vars_transposed.R [hunk 3]
```diff
@@ -151,20 +154,22 @@ derive_vars_transposed <- function(dataset,
...
+  value_var <- assert_symbol(enquo(value_var))
   assert_vars(by_vars)
   assert_data_frame(dataset, required_vars = replace_values_by_names(by_vars))
-  assert_data_frame(dataset_facm, required_vars = vars(!!!by_vars, FAGRPID, FATESTCD, FASTRESC))
+  assert_data_frame(dataset_facm, required_vars = vars(!!!by_vars, !!value_var, FAGRPID, FATESTCD))
 
   dataset %>%
     derive_vars_transposed(
-      select(dataset_facm, !!!unname(by_vars), FAGRPID, FATESTCD, FASTRESC),
+      select(dataset_facm, !!!unname(by_vars), !!unname(value_var), FAGRPID, FATESTCD),
```
Comment (bundfussr) [inline R/derive_vars_transposed.R:169]: `unname()` is not required for `value_var` because it isn't named anyway.

**Review by bundfussr — CHANGES_REQUESTED** [review]

**Comment by rossfarrugia** [comment]
@tamarasenior @bundfussr - can this one be approved and merged now? my Roche ADCM code fals over without it

**Review by bundfussr — APPROVED** [review]

**Comment by bundfussr** [comment]
@rossfarrugia , sorry, I missed to re-review. It is merged now.

**Pull request merged**